File: UI/eventsappadmin/lib/models/dogadjaji_report_response.dart
import 'package:json_annotation/json_annotation.dart';

part 'dogadjaji_report_response.g.dart';

@JsonSerializable()
class DogadjajiReportResponse {
  List<Map<String, Object>>? eventsByStatus;
  List<Map<String, Object>>? eventsByCategory;
  List<Map<String, Object>>? topSellingEvents;
  List<Map<String, Object>>? mostViewedEvents;
  List<Map<String, Object>>? mostSavedEvents;

  DogadjajiReportResponse(this.eventsByStatus, this.eventsByCategory,
      this.topSellingEvents, this.mostViewedEvents, this.mostSavedEvents);

  factory DogadjajiReportResponse.fromJson(Map<String, dynamic> json) =>
      _$DogadjajiReportResponseFromJson(json);

  Map<String, dynamic> toJson() => _$DogadjajiReportResponseToJson(this);
}

File: eventsApp.Model/DogadjajiReportResponse.cs
using System;
using System.Collections.Generic;
using System.Linq;
using System.Text;
using System.Threading.Tasks;

namespace eventsApp.Model
{
    public class DogadjajiReportResponse
    {
        public List<Dictionary<string,object>>? EventsByStatus { get; set; }
        public List<Dictionary<string, object>>? EventsByCategory { get; set; }
        public List<Dictionary<string, object>>? MostSavedEvents { get; set; }
        public List<Dictionary<string, object>>? MostViewedEvents { get; set; }
        public List<Dictionary<string, object>>? TopSellingEvents { get; set; }
    }
}
